server/user/mw.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const crypto = require('crypto');
+
+function getKeyFromRequest(req, res, next) {
+  const key = [req.ip, req.body.email].join(':');
+  req.key = crypto.createHash('sha256').update(key).digest('base64');
+  return next();
+}
+
+function resetLoginAttempts(requestLimiter) {
```

Comment:
Consider moving `loginRequestLimiter` to this file - it's a user-specific middleware so this looks like the place for it and we this factory can become a simple middleware.